src/main/java/de/rwth/idsg/steve/config/BeanConfiguration.java

```diff
@@ -82,13 +84,14 @@
 @ComponentScan("de.rwth.idsg.steve")
 public class BeanConfiguration implements WebMvcConfigurer {
 
-    private HikariDataSource dataSource;
+    @Autowired private HikariDataSource dataSource;
```

Comment:
You should remove the attribut and replace it by the bean factory method everywhere it is used (spring proxies are not supposed to instanciate more than once)